src/simpleinterfaces.cpp

```diff
@@ -18,41 +20,108 @@ using namespace std;
    ---------------------------------------------------------------------------
 */
 
+using f32  = float;
+using f64  = double;
+using c64  = std::complex<float>;
+using c128 = std::complex<double>;
+using i64  = BIGINT;
+
+void finufft_default_opts(finufft_opts *o) { finufft_default_opts_t(o); }
+void finufftf_default_opts(finufft_opts *o) { finufft_default_opts_t(o); }
+
+int finufft_makeplan(int type, int dim, const i64 *n_modes, int iflag, int ntrans,
+                     f64 tol, finufft_plan *pp, finufft_opts *opts) {
+  return finufft_makeplan_t<f64>(type, dim, n_modes, iflag, ntrans, tol,
+                                 reinterpret_cast<FINUFFT_PLAN_T<f64> **>(pp), opts);
+}
+int finufftf_makeplan(int type, int dim, const i64 *n_modes, int iflag, int ntrans,
+                      f32 tol, finufftf_plan *pp, finufft_opts *opts) {
+  return finufft_makeplan_t<f32>(type, dim, n_modes, iflag, ntrans, tol,
+                                 reinterpret_cast<FINUFFT_PLAN_T<f32> **>(pp), opts);
+}
+
+int finufft_setpts(finufft_plan p, i64 nj, f64 *xj, f64 *yj, f64 *zj, i64 nk, f64 *s,
+                   f64 *t, f64 *u) {
+  return finufft_setpts_t<f64>(reinterpret_cast<FINUFFT_PLAN_T<f64> *>(p), nj, xj, yj, zj,
+                               nk, s, t, u);
+}
+int finufftf_setpts(finufftf_plan p, i64 nj, f32 *xj, f32 *yj, f32 *zj, i64 nk, f32 *s,
+                    f32 *t, f32 *u) {
+  return finufft_setpts_t<f32>(reinterpret_cast<FINUFFT_PLAN_T<f32> *>(p), nj, xj, yj, zj,
+                               nk, s, t, u);
+}
+
+int finufft_execute(finufft_plan p, c128 *cj, c128 *fk) {
+  return finufft_execute_t<f64>(reinterpret_cast<FINUFFT_PLAN_T<f64> *>(p), cj, fk);
+}
+int finufftf_execute(finufftf_plan p, c64 *cj, c64 *fk) {
+  return finufft_execute_t<f32>(reinterpret_cast<FINUFFT_PLAN_T<f32> *>(p), cj, fk);
+}
+
+int finufft_destroy(finufft_plan p)
+// Free everything we allocated inside of finufft_plan pointed to by p.
+// Also must not crash if called immediately after finufft_makeplan.
+// Thus either each thing free'd here is guaranteed to be nullptr or correctly
+// allocated.
+{
+  if (!p) // nullptr ptr, so not a ptr to a plan, report error
+    return 1;
+
+  delete reinterpret_cast<FINUFFT_PLAN_T<f64> *>(p);
```

Comment:
The cast is needed because the C data type to which `p` points is not defined at all. All the compiler knows is
that `p` points to an object of type `finufft_plan_s`, but we never say what's in there.
(Well, we define it somewhere in `defs.h`, but that's only included in tests from now on.)